app/src/main/java/com/codingbad/vpntoggle/model/ApplicationItem.java

```diff
@@ -77,6 +77,15 @@ public void setApplicationName(String applicationName) {
         this.applicationName = applicationName;
     }
 
+    public boolean equals(Object object) {
+        if (!(object instanceof ApplicationItem))
+            return false;
+        if (object == this)
+            return true;
+        ApplicationItem other = (ApplicationItem) object;
+        return other.getUID() == this.getUID();
+    }
+
```

Comment:
@xeroxnir I would change it a little:

public boolean equals(Object object) {
        if (object == this)
            return true;
       return object instanceof ApplicationItem && ((ApplicationItem) object).getUID() == this.getUID();
}